R/GraphLearner.R

```diff
@@ -102,6 +94,37 @@ GraphLearner = R6Class("GraphLearner", inherit = Learner,
         return(value$clone(deep = TRUE))
       }
       value
+    },
+    get_predict_type = function() {
+      # recursively walk backwards through the graph 
+      get_po_predict_type = function(x) {
+        if (!is.null(x$predict_type)) return(x$predict_type)
+        prdcssrs = self$graph$edges[dst_id == x$id, ]$src_id
+        if (length(prdcssrs)) {
+          # all non-null elements
+          predict_types = keep(map(self$graph$pipeops[prdcssrs], get_po_predict_type), Negate(is.null))
+          if (length(unique(predict_types)) == 1L)
```

Comment:
its a bit weird to ignore NULL-returns by `get_po_predict_type` calls, but *also* encode inconsistent predecessor predict types as `NULL`.

Reply:
true